lib/core/locale.nit

```diff
@@ -0,0 +1,64 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Enumeration of common locale names as well as default locale name
+module locale is pkgconfig ("icu-io", "icu-i18n", "icu-uc")
+
+`{
+	#include <string.h>
+	#include <unicode/uloc.h>
+`}
+
+private fun get_default_locale(locale: CString): Int `{
+	const char * default_locale = uloc_getDefault();
+	if (locale != NULL) {
+		strcpy(locale, default_locale);
+	}
+	return strlen(default_locale);
+`}
+
+redef class Sys
+	fun default_locale: String do
+		var required_length = get_default_locale(new CString.nul)
+		var locale = new CString(required_length + 1)
+		get_default_locale(locale)
+		return locale.to_s
+	end
+
+	fun chinese_locale: String do return "zh"
+	fun english_locale: String do return "en"
+	fun french_locale: String do return "fr"
+	fun german_locale: String do return "de"
+	fun italian_locale: String do return "it"
+	fun japanese_locale: String do return "ja"
+	fun korean_locale: String do return "ko"
+	fun simplified_chinese_locale: String do return "zh_CN"
+	fun traditional_chinese_locale: String do return "zh_TW"
+	fun canada_locale: String do return "en_CA"
+	fun canada_french_locale: String do return "fr_CA"
+	fun china_locale: String do return "zh_CN"
+	fun france_locale: String do return "fr_FR"
+	fun germany_locale: String do return "de_DE"
+	fun italy_locale: String do return "it_IT"
+	fun japan_locale: String do return "ja_JP"
+	fun korea_locale: String do return "ko"
+	fun taiwan_locale: String do return "zh_TW"
+	fun uk_locale: String do return "en_GB"
+	fun us_locale: String do return "en_US"
+end
+
+redef class CString
+	# Returns a null `char *`
+	new nul `{ return NULL; `}
```

Comment:
If you define the same property in two places, it might be suitable to move it in a shared class.